app/com/linkedin/drelephant/tez/fetchers/TezFetcher.java

```diff
@@ -255,31 +263,35 @@ private long getDagEndTime(URL url) throws IOException, AuthenticationException
     }
 
     private void getTaskDataAll(URL vertexListUrl, String dagId, List<TezTaskData> mapperList,
-                  List<TezTaskData> reducerList) throws IOException, AuthenticationException {
+                                List<TezTaskData> reducerList, List<TezTaskData> scopeTaskList) throws IOException, AuthenticationException {
 
       JsonNode rootVertexNode = ThreadContextMR2.readJsonNode(vertexListUrl);
       JsonNode vertices = rootVertexNode.path("entities");
       boolean isMapVertex = false;
-
+      String vertexType = null;
       for (JsonNode vertex : vertices) {
         String vertexId = vertex.get("entity").getTextValue();
         String vertexClass = vertex.path("otherinfo").path("processorClassName").getTextValue();
 
         if (vertexClass.equals("org.apache.hadoop.hive.ql.exec.tez.MapTezProcessor"))
-          isMapVertex = true;
+          vertexType = VertexType.MAPPER.name();
         else if (vertexClass.equals("org.apache.hadoop.hive.ql.exec.tez.ReduceTezProcessor"))
-          isMapVertex = false;
+          vertexType = VertexType.REDUCER.name();
+        else if (vertexClass.equals("org.apache.pig.backend.hadoop.executionengine.tez.runtime.PigProcessor"))
+          vertexType = VertexType.SCOPE.name();
 
         URL tasksByVertexURL = getTaskListByVertexURL(dagId, vertexId);
-        if(isMapVertex)
-          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, mapperList, true);
-        else
-          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, reducerList, false);
+        if(vertexType.equals(VertexType.MAPPER.name()))
+          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, mapperList,vertexType);
+        else if (vertexType.equals(VertexType.REDUCER.name()))
+          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, reducerList, vertexType);
+        else if (vertexType.equals(VertexType.SCOPE.name()))
+          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, scopeTaskList, vertexType);
```

Comment:
I think you can move this method call to line 281, and get this if check on line 288.